articles/understanding-systemd/index.md

```diff
@@ -0,0 +1,291 @@
+systemd is capable of starting services in parallel which speeds up the boot process. It is also capable of handling hot-plug devices; such as starting up services on demand at the occurrence of an event (like upstart). The process of managing service daemons in System V  is a little bit involved. This is so because scripts get detached from the daemons they spawn upon runtime. Some daemons spawn 3rd party processes that get orphaned and re-parented by PID 1. Some processes even re-fork and rename themselves which makes process management difficult.
+
+systemd makes daemon/service management easier by placing their processes in a control group (cgroup).
+
+Here, spawned processes are made members of their parent cgroup (named after the service). These processes are not capable of leaving their cgroup until their parent terminates them. This prevents the creation of zombie processes. It also eases the management (running kill -9 as an example) of related processes since they belong to one cgroup.
+
+systemd also provides a tool called "***systemctl***". ***man 1 systemctl*** says ***systemctl*** may be used to introspect and control the state of the "***systemd***" system and service manager.
+
+This tool can be used to start and stop services including other things as we will soon see. System V init implementations manage system state by the use of shell scripts. 
+systemd on the flip-side takes a different approach towards writing configuration files. It does this by expressing commands in a declarative format. This makes the job of administering Linux systems easier by making uniform all aspects of system administration.
+
+
+[ULSAH](https://www.amazon.com/UNIX-Linux-System-Administration-Handbook/dp/0134277554) gives a succinct description on systemd's mode of operation:
+
+*Much like a package management system, systemd defines a robust dependency model, not only among services but also among "targets," systemd's term for the operational modes that traditional init calls runlevels. systemd not only manages processes in parallel, but also manages network connections (networkd), kernel log entries (journald), and logins (logind).*
+
+systemd is also compatible with existing init scripts of System V.
+
+---
+
+### systemd deep-dive
+During boot time systemd loads its configuration, determines its boot goal and dependences. It then activates them afterward.
+
+systemd also doesn't follow a sequential pattern for starting services. Rather, a flexible process by which extra components activate when system events occur.
+
+The systemd Linux manual states that systemd provides a dependency system between various entities called "units".
+Units encapsulate various objects that are relevant for system boot-up and maintenance. A **unit type** is a type of task relevant to the management of a Linux system. A task is a unit in systemd parlance. These units are a standardized representation of system resources managed by the suite of systemd daemons. 
```

Comment:
A hyphen is missing in 'task relevant', consider changing it to 'task-relevant'.

Reply:
This is a type of task that's relevant to Linux system management.
In this case 'task relevant' are separate words.

Reply:
I know but consider making it stand out, put more emphasis on it, for example, make it bold or something..

Reply:
Okay, it's much more distinctive now.